core/domain/learner_group_fetchers.py

```diff
@@ -107,7 +107,9 @@ def can_multi_learners_share_progress(
 
     progress_sharing_permissions: List[bool] = []
     for model in learner_group_user_models:
-        # Ruling out the possibility of None for mypy type checking.
+        # Ruling out the possibility of None for mypy type checking, because
```

Comment:
Not sure I buy this. If it happens in prod for any reason (maybe data corruption), should we raise exception or handle this gracefully? @paprajapati9 @aks681

Reply:
Hmm... on second thought we should do that. @sahiljoster32 can you please raise an exception saying `Expected learner group user model to exist for the given user_id` if model is None. Thanks!

Reply:
This one is done but there are some other asserts that are introduced by you -- in PR https://github.com/oppia/oppia/pull/15840

https://github.com/oppia/oppia/blob/dc57146488ae724111b012e5d5c88d06faf537a0/core/domain/learner_group_services.py#L616
https://github.com/oppia/oppia/blob/dc57146488ae724111b012e5d5c88d06faf537a0/core/domain/learner_group_services.py#L541

can you please give a reason or confirm that exception is expected here !!!

Reply:
@paprajapati9 PTAL!

Reply:
Yes @sahiljoster32, please add an exception saying `Expected learner group user model to exist for the given learner invited to join the learner group` at both places. Thanks!

Reply:
Done!